classes/validation.php

```diff
@@ -884,11 +884,22 @@ public function _validation_valid_url($val)
 	 * Validate IP using PHP's filter_var()
 	 *
 	 * @param   string  $val
+	 * @param   string  ipv4|ipv6
 	 * @return  bool
 	 */
-	public function _validation_valid_ip($val)
+	public function _validation_valid_ip($val, $flag = null)
 	{
-		return $this->_empty($val) || filter_var($val, FILTER_VALIDATE_IP);
+		switch (strtolower($flag))
+		{
+			case 'ipv4':
+				$flag = FILTER_FLAG_IPV4;
+				break;
+			case 'ipv6':
+				$flag = FILTER_FLAG_IPV6;
+				break;
```

Comment:
perhaps add a
````
default:
	$flag = null;
`````
to avoid all sorts of funny errors if some other flag is passed? Or an exception with "invalid flag" or something"?